pkg/apis/batch/types.go

```diff
@@ -138,6 +138,16 @@ type JobSpec struct {
 
 	// Describes the pod that will be created when executing a job.
 	Template api.PodTemplateSpec
+
+	// ttlSecondsAfterFinished limits the lifetime of a Job that has finished
+	// execution (either Complete or Failed). If this field is set, once the Job
+	// finishes, it will be deleted after ttlSecondsAfterFinished expires. If
+	// this field is unset, the Job will not be cleaned up after it finishes. If
```

Comment:
Might be worth mentioning that it honors Finalizers.

Reply:
Updated API docs.